lib/foreman/advisory_lock_manager.rb

```diff
@@ -45,7 +47,7 @@ def with_session_lock(lock_name)
       # The method can result in different results with different Ruby versions, but this is
       # unlikely to happen (possibly during rolling restart when upgrading to new Ruby).
       def generate_lock_id(lock_name)
-        lock_name.to_s.hash & 0x7FFFFFFFFFFFFFFF
+        Digest::SHA1.digest(lock_name).unpack1('q')
```

Comment:
Is SHA1 available on FIPS-enforced machines?

Reply:
Looks like that, I enabled FIPS on my CentOS 7 installation (following [this tutorial](https://www.thegeekdiary.com/how-to-make-centos-rhel-7-fips-140-2-compliant/)):
```
# irb
irb(main):001:0> require 'digest/sha1'
=> true
irb(main):002:0> require 'digest/md5'
=> true
irb(main):003:0> Digest::SHA1.digest "Hello World"
=> "\nMU\xA8\xD7x\xE5\x02/\xABp\x19w\xC5\xD8@\xBB\xC4\x86\xD0"
irb(main):004:0> Digest::MD5.digest "Hello World"
md5_dgst.c(82): OpenSSL internal error, assertion failed: Digest MD5 forbidden in FIPS mode!
Aborted
```

Reply:
I think on CentOS 8 it may be excluded. I'd check with that as well.

Also, the comment above it is out of date if SHA is used instead of built in hashing.

Reply:
Interesting, on centos8 both seem to work :thinking: 
```
[root@centos8-katello-nightly ~]# cat /proc/sys/crypto/fips_enabled
1
[root@centos8-katello-nightly ~]# foreman-rake console
Loading production environment (Rails 6.0.3.5)
irb(main):001:0> require 'digest/sha1'
=> false
irb(main):002:0> require 'digest/md5'
=> false
irb(main):003:0> Digest::SHA1.digest "Hello World"
=> "\nMU\xA8\xD7x\xE5\x02/\xABp\x19w\xC5\xD8@\xBB\xC4\x86\xD0"
irb(main):004:0> Digest::MD5.digest "Hello World"
=> "\xB1\n\x8D\xB1d\xE0uA\x05\xB7\xA9\x9B\xE7.?\xE5"
```

Reply:
That's very surprising and doesn't sound right. Could be a bug in EL8's Ruby when it comes to FIPS compliance.

Reply:
Could be :shrug: 

So should I rather use `Digest::SHA512` then?
I do only use as much data as needed for the 64bit Integer after all.

Reply:
I think SHA256 is good enough here. It probably doesn't matter that much since it's not very performance sensitive, but it is executed on every startup so I'd at least think about it a little.

@tbrisker thoughts?

Reply:
Performance of all functions seem to be fast enough to not be noticeable when done just once during startup:
```ruby
[45] pry(main)> Benchmark.ips do |x|
[45] pry(main)*   x.report('sha1'){Digest::SHA1.digest('db_seed')}
[45] pry(main)*   x.report('sha256'){Digest::SHA256.digest('db_seed')}  
[45] pry(main)*   x.report('sha512'){Digest::SHA512.digest('db_seed')}  
[45] pry(main)*   x.report('md5'){Digest::MD5.digest('db_seed')}  
[45] pry(main)*   x.compare!
[45] pry(main)* end  
Warming up --------------------------------------
                sha1    49.687k i/100ms
              sha256    44.472k i/100ms
              sha512    32.756k i/100ms
                 md5    50.187k i/100ms
Calculating -------------------------------------
                sha1    468.085k (±16.4%) i/s -      2.286M in   5.022709s
              sha256    374.576k (±15.6%) i/s -      1.868M in   5.126641s
              sha512    288.907k (±31.5%) i/s -      1.277M in   5.078512s
                 md5    481.065k (±16.7%) i/s -      2.359M in   5.052414s

Comparison:
                 md5:   481064.7 i/s
                sha1:   468085.0 i/s - same-ish: difference falls within error
              sha256:   374575.9 i/s - same-ish: difference falls within error
              sha512:   288906.8 i/s - 1.67x  (± 0.00) slower
```
Considering that, i would rather err on choosing a slower, stronger algorithm that will hopefully be FIPS-compliant for a longer time, but I don't have a strong preference.